packages/react-server/core/context/Navigator.js

```diff
@@ -50,7 +50,7 @@ class Navigator extends EventEmitter {
 
 		this._haveInitialized = true;
 
-		var route = this.router.getRoute(request.getUrl(), {navigate: {path:request.getUrl(), type:type}});
+		var route = this.router.getRoute(request.getUrl(), {method: request.getMethod()});
```

Comment:
I'm confident in all of the changes here except this `type` thing.  I really don't understand it at all.